src/assessments/seo/textLengthAssessment.js

```diff
@@ -41,127 +68,68 @@ class TextLengthAssessment extends Assessment {
 	 *
 	 * @param {Paper} paper The Paper object to assess.
 	 * @param {Researcher} researcher The Researcher object containing all available researches.
-	 * @param {object} i18n The locale object.
+	 * @param {Object} i18n The locale object.
 	 *
 	 * @returns {AssessmentResult} The result of the assessment, containing both a score and a descriptive text.
 	 */
 	getResult( paper, researcher, i18n ) {
-		let wordCount = researcher.getResearch( "wordCountInText" );
+		this.wordCount = researcher.getResearch( "wordCountInText" );
 		let assessmentResult = new AssessmentResult();
+		const calculatedResult = this.calculateResult();
 
-		assessmentResult.setScore( this.calculateScore( wordCount ) );
+		assessmentResult.setScore( calculatedResult.score );
 		assessmentResult.setText(
-			i18n.sprintf( this.translateScore( assessmentResult.getScore(), wordCount, i18n ), wordCount, this._config.recommendedMinimum ) );
+			this.translateScore( calculatedResult.resultText, calculatedResult.resultTextPlural, i18n ) );
```

Comment:
As per previous comment regarding this: Consider splitting this over multiple lines.